coremltools/converters/nnssa/coreml/ssa_converter.py

```diff
@@ -683,6 +684,8 @@ def __compare_propagated_and_inferred_shape(self, name, type_):
             for ashape in type_.T:
                 assert ashape.get_shape() == element_shape
             inferred_shape = [-1] + list(element_shape)
+        elif builtins.is_tuple(type_):
+            inferred_shape = type_.T[0].get_shape()
```

Comment:
Question: Why are we assigning the shape of first tensor in the tuple i.e why index "0"?

Reply:
@DawerG san,

I have no idea from the view point of specifications of tuple type.
In coremltools codes, tuple constructors basically seem to get one argument.
So index 0 is used.